index.bs

```diff
@@ -2219,6 +2255,88 @@ To <dfn>parse trigger specs</dfn> given a [=map=] |map|, a [=moment=]
+1. If |pendingSource|'s [=attribution source/attribution scope limit=] is null, return.
+1. Let |sourcesToRemove| be a new [=list/is empty|empty=] [=list=].
+1. Let |cache| be the user agent's [=attribution source cache=].
+1. [=set/iterate|For each=] |source| of |cache|:
+    1. Let |commonDestinations| be the [=set/intersection=] of |source|'s [=attribution source/attribution destinations=] and |pendingSource|'s [=attribution source/attribution destinations=].
+    1. If |commonDestinations| [=set/is empty=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/source origin=] and |pendingSource|'s [=attribution source/source origin=] are not [=same origin=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/attribution scope limit=] is null, [=iteration/continue=].
+    1. If |source|'s [=attribution source/attribution scope limit=] is greater than or equal to |pendingSource|'s [=attribution source/attribution scope limit=], [=iteration/continue=].
+    1. [=list/Append=] |source| to |sourcesToRemove|.
+1. [=set/Remove=] |sourcesToRemove| from |cache|.
+
+To <dfn>remove attribution scopes for sources with attribution scopes not selected if necessary</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |attributionScopeLimit| be |pendingSource|'s [=attribution source/attribution scope limit=].
+1. If |attributionScopeLimit| is null, return.
+1. Let |cache| be the user agent's [=attribution source cache=].
+1. [=list/iterate|For each=] |destination| in |pendingSource|'s [=attribution source/attribution destinations=]:
+    1. Let |matchingSources| be a new [=list/is empty|empty=] [=list=].
+    1. [=set/iterate|For each=] |source| of |cache|:
+        1. If |source|'s [=attribution source/source origin=] and |pendingSource|'s [=attribution source/source origin=] are not [=same origin=], [=iteration/continue=].
+        1. If |source|'s [=attribution source/attribution destinations=] does not [=set/contains|contain=] |destination|, [=iteration/continue=].
+        1. [=list/Append=] |source| to |matchingSources|.
+    1. Set |matchingSources| to the result of [=list/sort in descending order|sorting=] |matchingSources|
+        in descending order, with |a| being less than |b| if
+        |a|'s [=attribution source/source time=] is less than |b|'s [=attribution source/source time=].
+    1. Let |selectedAttributionScopes| be a [=set=].
+    1. If |pendingSource|'s [=attribution source/attribution scopes=] is not null:
+        1. Set |selectedAttributionScopes| to |pendingSource|'s [=attribution source/attribution scopes=].
+    1. [=set/iterate|For each=] |source| of |matchingSources|:
+        1. If |source|'s [=attribution source/attribution scopes=] is null, [=iteration/continue=].
+        1. Let |newSelectedAttibutionScopes| be the [=set/union=] of |selectedAttributionScopes| and |source|'s [=attribution source/attribution scopes=].
```

Comment:
This is being declared anew in each iteration. Is it supposed to be cumulative, rather?

Reply:
My bad, it's reassigned later.